File: backend/src/main/java/com/coffeeshop/backend/payments/Payment.java
package com.coffeeshop.backend.payments;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

public record Payment(
        UUID paymentId,
        String storeId,
        CoffeeType coffeeType,
        BigDecimal price,
        String currency,
        String loyaltyCardId,
        String idempotencyKey,
        String remotePaymentId,
        Instant createdAt
) {}

File: client/src/domain/payment/Payment.js
export class Payment {
  constructor({ storeId, idempotencyKey, coffeeType, price, currency, loyaltyCardId, rowIndex }) {
    this.storeId = storeId
    this.idempotencyKey = idempotencyKey
    this.coffeeType = coffeeType
    this.price = price
    this.currency = currency
    this.loyaltyCardId = loyaltyCardId
    this.rowIndex = rowIndex
  }

  toApiBody() {
    return {
      coffeeType: this.coffeeType,
      price: Number(this.price),
      currency: this.currency,
      loyaltyCardId: this.loyaltyCardId,
    }
  }
}
